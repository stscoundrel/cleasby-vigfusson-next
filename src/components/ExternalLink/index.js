export default function ExternalLink({ href, title }) {
  return (
    <a
      href={href}
      target="_blank"
      rel="noopener noreferrer"
    >
      {title}
    </a>
  )
}
