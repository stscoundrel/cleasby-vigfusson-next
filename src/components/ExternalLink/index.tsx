interface ExternalLinkProps{
  href: string,
  title: string,
}

export default function ExternalLink({ href, title }: ExternalLinkProps) {
  return (
    <a
      href={href}
      target="_blank"
      rel="noopener noreferrer"
    >
      {title}
    </a>
  )
}
