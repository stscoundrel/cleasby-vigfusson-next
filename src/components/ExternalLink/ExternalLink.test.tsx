import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import ExternalLink from './index'

describe('ExternalLink component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<ExternalLink title="Other site" href="www.takemeontheotherside.com" />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<ExternalLink title="Other site" href="www.takemeontheotherside.com" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Has correct title', () => {
    const tree = renderer.create(<ExternalLink title="Other site" href="www.takemeontheotherside.com" />)
    const { root } = tree

    expect(root.findByType('a').children).toEqual(['Other site'])
  })

  test('Has correct url', () => {
    const tree = renderer.create(<ExternalLink title="Other site" href="www.takemeontheotherside.com" />).toJSON()

    expect(tree.props.href).toEqual('www.takemeontheotherside.com')
  })
})
