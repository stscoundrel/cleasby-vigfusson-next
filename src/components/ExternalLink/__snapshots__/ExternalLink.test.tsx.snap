// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink component Matches snapshot 1`] = `
<a
  href="www.takemeontheotherside.com"
  rel="noopener noreferrer"
  target="_blank"
>
  Other site
</a>
`;
